packages/shorebird_cli/lib/src/shorebird_flavor_mixin.dart

```diff
@@ -0,0 +1,115 @@
+import 'dart:io';
+
+import 'package:collection/collection.dart';
+import 'package:path/path.dart' as p;
+import 'package:platform/platform.dart';
+import 'package:shorebird_cli/src/command.dart';
+
+mixin ShorebirdFlavorMixin on ShorebirdCommand {
+  Future<Set<String>> extractProductFlavors(
+    String path, {
+    Platform platform = const LocalPlatform(),
+  }) async {
+    final executable = platform.isWindows ? 'gradlew.bat' : 'gradlew';
+    final androidStudioPath = _getAndroidStudioPath(platform);
+    final javaPath = androidStudioPath != null
+        ? _getJavaPath(androidStudioPath, platform)
+        : null;
+    final result = await process.run(
+      p.join(path, 'android', executable),
+      ['app:tasks', '--all', '--console=auto'],
+      runInShell: true,
+      workingDirectory: p.join(path, 'android'),
+      environment: {
+        if (javaPath != null) 'JAVA_HOME': javaPath,
+      },
+    );
+
+    if (result.exitCode != 0) throw Exception('${result.stderr}');
+
+    final variants = <String>{};
+    final assembleTaskPattern = RegExp(r'assemble(\S+)');
+    for (final task in '${result.stdout}'.split('\n')) {
+      final match = assembleTaskPattern.matchAsPrefix(task);
+      if (match != null) {
+        final variant = match.group(1)!.toLowerCase();
+        if (!variant.endsWith('test')) {
+          variants.add(variant);
+        }
+      }
+    }
+
+    final productFlavors = <String>{};
+    for (final variant1 in variants) {
```

Comment:
A comment explaining what this is doing and why would be helpful

Reply:
Yup good call just added ✅